bumble/controller.py

```diff
@@ -623,32 +994,42 @@ def on_classic_role_change(self, peer_address, new_role):
     # Advertising support
     ############################################################
     def on_advertising_timer_fired(self):
-        self.send_advertising_data()
-        self.advertising_timer_handle = asyncio.get_running_loop().call_later(
-            self.advertising_interval / 1000.0, self.on_advertising_timer_fired
-        )
+        self.advertising_timer_handle = None
 
-    def start_advertising(self):
-        # Stop any ongoing advertising before we start again
-        self.stop_advertising()
+        self.send_advertising_data()
 
-        # Advertise now
-        self.advertising_timer_handle = asyncio.get_running_loop().call_soon(
-            self.on_advertising_timer_fired
+        # Compute the time of the next advertisement
+        next_advertisement = min(
```

Comment:
nit: This could be replaced with a heapq
```py

# Advertiser
def send_advertising_data(self):
  ...
  heappush(timer_heap, (time.time()+interval, self))
  
def on_advertising_timer_fired(self):
  next_advertisement, advertiser = heappop(timer_heap)
  advertiser.send_advertising_data()
```

Reply:
Done